File: proyecto1/vuelos.cpp
#include "vuelos.h"

Vuelos::Vuelos()
{
    ciudadOrigen = "CLO";
    estado = true;
}

Vuelos::Vuelos(int id, const string &fecha, const string &ciudadDestino, const string &hora)
    : identificacion(id), fecha(fecha),
      ciudadDestino(ciudadDestino), hora(hora),
      numPasajeros(0)
{
    ciudadOrigen = "CLO";
    estado = true;
}

void Vuelos::agregarPasajero()
{
    if (numPasajeros < capacidad)
        numPasajeros++;
    else
        printf("Vuelo lleno\n");
    if (numPasajeros == capacidad)
        estado = false;
}

void Vuelos::printVuelo()
{
    cout << "Fecha: " << fecha << endl
         << "Hora " << hora << endl
         << "Ciudad de origen: " << ciudadOrigen << endl
         << "Ciudad de destino " << ciudadDestino << endl;
}

bool Vuelos::disponible()
{
    return estado;
}
/*
Vuelos obtenerDatosVuelo()
{
    int id, capacidad, numPasajeros;
    string fecha, ciudadOrigen, ciudadDestino, hora, ej;

    try
    {
        cout << "Ingrese la identificacion del vuelo: ";
        cin >> ej;
        id = stoi(ej);
    }
    catch (const invalid_argument &e)
    {
        cerr << "Error argumento invalido " << e.what() << "Ingrese un numero entero" << endl;
        cin >> id;
    }

    cin.ignore(); // Limpiar el buffer de entrada

    cout << "Ingrese la fecha del vuelo (YYYY-MM-DD): ";
    getline(cin, fecha);

    cout << "Ingrese la ciudad de destino: ";
    getline(cin, ciudadDestino);

    cout << "Ingrese la hora del vuelo (HH:MM): ";
    getline(cin, hora);

    // Crear y devolver un objeto Vuelos con los datos ingresados
    Vuelos tmp(id, fecha, ciudadDestino, hora);
    return tmp;
} */

File: proyecto2/vuelos.py
""" from tripulante import Tripulante
from pasajero import Pasajero """


class Vuelos:
    def __init__(self, id, fecha, ciudad_destino, hora,aerolinea):
        self.identificacion = id
        self.capacidad = 100  # Puedes establecer la capacidad deseada
        self.num_pasajeros = 0
        self.fecha = fecha
        self.ciudad_origen = "CLO"
        self.ciudad_destino = ciudad_destino
        self.hora = hora
        self.tripulantes = []
        self.tr = []
        self.estado = True
        self.aerolinea = aerolinea

    def agregarPasajero(self):
        if self.num_pasajeros < self.capacidad:
            self.num_pasajeros += 1
        else:
            self.estado = False
    
    def agregarTripulante(self, tripulantes, tr):
        self.tripulantes = tripulantes
        self.tr = tr

    def printVuelo(self):
        data = {"Aerolinea": self.aerolinea,"Ciudad de origen:": self.ciudad_origen,"Ciudad de destino:": self.ciudad_destino,"Fecha:": self.fecha, "Hora:": self.hora}
        if self.estado:
            data["Disponibilidad:"] = "disponible"
        else:
            data["Disponibilidad:"] = "Vuelo lleno"
        return data
    
    def printTripulantes(self):
        return self.tr
    def disponible(self):
        return self.estado
